src/app/SharpRaven/RavenClient.cs

```diff
@@ -363,6 +393,19 @@ protected virtual string Send(JsonPacket packet)
             }
             catch (Exception exception)
             {
+                if (!cachePacketsWhenOffline) return HandleException(exception, requester);
+                try
+                {
+                    if (!string.IsNullOrEmpty(packet.Message)) packet.Message += "\n";
+                    packet.Message +=
+                        $"Attempted to send on {DateTime.UtcNow.ToShortDateString()} {DateTime.UtcNow.ToLongTimeString()} UTC however packet failed to send and was cached locally.";
+                    var bytes = packet.SerialiseBytes();
+                    var fileName =
+                        $"{DateTime.UtcNow.ToShortDateString().Replace("/", "-")} {DateTime.UtcNow.ToLongTimeString().Replace(":", "-")} - {GetHashCode()}.json";
+                    var path = Path.Combine(System.Environment.CurrentDirectory, fileName);
```

Comment:
I think the folder should be based around the DSN, a temporary folder, or something similar. In a web context, `CurrentDirectory` might not be writable.